### PR TITLE
Add different recommendations and remove relatedProductsQuery

### PR DESCRIPTION
**Pull request by estacioneto**
#### What is the purpose of this pull request?
- Related Products can be view, buy or similars recommendations.
- No need of GraphQL query. The information comes from the context API.

#### What problem is this solving?
- Only one recommendation.
- Unnecessary query.

#### How should this be manually tested?

[Access the workspace](https://estacio--storecomponents.myvtex.com/admin/cms/storefront)

#### Screenshots or example usage

image
image
image

#### Types of changes
- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: react/RelatedProducts.js [hunk 1]
```diff
@@ -2,81 +2,86 @@ import './global.css'
...
 /**
  * Related Products Component. Queries and shows the related products
  */
-class RelatedProducts extends Component {
+export default class RelatedProducts extends Component {
+  static propTypes = {
+    /** Main product to have related products queried */
+    slug: PropTypes.string,
+    /** Graphql productQuery response. */
+    productQuery: PropTypes.shape({
```
Comment (ClaudivanFilho) [inline react/RelatedProducts.js]: Não seria melhor o RelatedProducts receber já o recomendations direto?

Comment (estacioneto) [reply react/RelatedProducts.js]: O contexto (store) que provê a todos os elementos colocados no extension container da página de produto todos os dados da `productQuery`, por isso recebe a query completa

File: react/RelatedProducts.js [hunk 2]
```diff
@@ -2,81 +2,86 @@ import './global.css'
...
-      (data &&
-        !data['error'] &&
-        data.product &&
-        data.product.recommendations &&
-        data.product.recommendations.view) ||
+      (productQuery &&
+        !productQuery['error'] &&
+        productQuery.product &&
+        productQuery.product.recommendations &&
+        productQuery.product.recommendations[recommendation]) ||
```
Comment (ClaudivanFilho) [inline react/RelatedProducts.js]: acho melhor usar o `path` do ramda aqui

File: react/RelatedProducts.js [hunk 3]
```diff
@@ -1,82 +1,85 @@
...
+    const { productQuery, productList } = this.props
+    const recommendation = this.props.recommendation.split('.').pop()
     const products =
-      (data &&
-        !data['error'] &&
-        data.product &&
-        data.product.recommendations &&
-        data.product.recommendations.view) ||
+      (!path(['error'], productQuery) &&
+        path(['product', 'recommendations', recommendation], productQuery)) ||
```
Comment (ClaudivanFilho) [inline react/RelatedProducts.js]:
```
 (!path(['error'], productQuery) && 
    path(['product', 'recommendations', recommendation], productQuery)) 
```
acho que só isso bastaria :arrow_down: , já que não é possível vir error e product ao mesmo tempo na query
```
path(['product', 'recommendations', recommendation], productQuery)) 
```

**Pull request merged**